### ISSUE TITLE
Doesn't appear to work on scripts

### ISSUE THREAD

**Issue by HackAttack** [CodeNarc/CodeNarc#152]
Running CodeNarc 24.1 via Gradle 2.12, when I have a Groovy script (i.e. no class declaration), no checks seem to run on this file. When I convert the script into a class with a main method, CodeNarc correctly detects errors.

**Comment by chrismair** [comment]
Does the script file have a *.groovy extension?

Are you able to see the Log4J output? Is it possible CodeNarc is encountering a compilation error or some other error for the script?

**Comment by HackAttack** [comment]
It is actually complaining about compilation failures for all of my files, even though they do compile and the application works. I get the following message for each of my files:
```
Compilation failed for [CustomCompilerPhaseSourceDecorator[SourceFile[/path/to/file.groovy]]]
```
I actually get another error only for the problematic file:
```
Error from [org.codenarc.rule.naming.VariableNameRule] processing source file [/path/to/Deploy.groovy]
Error processing filePath: 'Deploy.groovy'
```
I will work on a reproduction.

**Comment by HackAttack** [comment]
The script thing was sort of a red herring but sort of not because it’s related to the `@Field` annotation, which is only used in scripts. I’ve narrowed the issue—it’s fine with the `@Field` annotation _or_ generics, but not both.
```groovy
// OK
Map<String, String> env = new HashMap<String, String>(System.getenv())

// OK
@Field
Map env = new HashMap<String, String>(System.getenv())

// Fails to process
@Field
Map<String, String> env = new HashMap<String, String>(System.getenv())
```
I am still getting bogus “compilation failed” messages, but that‘s neither here nor there.

**Comment by chrismair** [comment]
Do you have any "enhanced rules" in your ruleset, but without having the necessary classes on your classpath?
http://codenarc.sourceforge.net/codenarc-rules-enhanced.html

That could explain the compilation failed messages. If so, remove those rules from your ruleset.

**Comment by chrismair** [comment]
Can you post the full error you are getting from the "problematic" file? And is that only for the VariableNameRule?

**Comment by chrismair** [comment]
I was able to reproduce and fix an ArrayIndexOutOfBoundsException when a variable name had an annotation. The annotation messed up the line number for the source line. 

It looks like what happened is: The generics in your failing example just pushed the index of the variable name beyond the line length of the previous line (with the annotation), so the substring failed with ArrayIndexOutOfBoundsException.

If that is the exception you saw, let me know, and I'll close this issue.

**Comment by HackAttack** [comment]
Thank you, that is the same bug that I’m seeing. And thanks for the pointer about the enhanced classpath rules.

**Comment by chrismair** [comment]
Thanks. Fixed for CodeNarc 0.25.2
